src/server/snapshot.cc

```diff
@@ -337,7 +346,13 @@ size_t SliceSnapshot::FlushSerialized(SerializerBase::FlushState flush_state) {
     return 0;
 
   uint64_t id = rec_id_++;
-  DVLOG(2) << "Pushing " << id;
+
+  if (use_background_mode_) {
```

Comment:
why is it important to yield here? why not after the call `consumer_->ConsumeData` below?